From: Misael Lopez Cruz <misael.lopez@ti.com>
Date: Fri, 21 Jan 2011 20:34:37 +0000 (-0600)
Subject: ASoC: ABE DSP: Use 98MHz for ABE OPP25 processing
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=e9c656e4d51ade94dc286dd747a1b8b902d697e3;hp=85530db290a41ced7bc00ff1b15ec80bdd102fcf

ASoC: ABE DSP: Use 98MHz for ABE OPP25 processing

A background noise is present in headset path during
audio playback at OPP50_98 (49MHz aess clock). That
noise is eliminated by using 98MHz aess clock and keeping
ABE at OPP25 processing. This solution is temporal and
should be reverted once noise root cause is fixed.

Change-Id: If1048885c62703e30516f03d7139aa2e944bb768
Signed-off-by: Misael Lopez Cruz <misael.lopez@ti.com>
---

Index: android-2.6.35/sound/soc/omap/omap-abe-dsp.c
===================================================================
--- android-2.6.35.orig/sound/soc/omap/omap-abe-dsp.c	2011-01-25 11:36:13.357812017 +0530
+++ android-2.6.35/sound/soc/omap/omap-abe-dsp.c	2011-01-25 11:47:33.147811916 +0530
@@ -1668,7 +1668,7 @@
 		case 25:
 			abe_set_opp_processing(ABE_OPP25);
 			udelay(250);
-			omap_device_set_rate(&pdev->dev, &pdev->dev, 49000000);
+			omap_device_set_rate(&pdev->dev, &pdev->dev, 98000000);
 			break;
 		case 50:
 		default:
@@ -1681,7 +1681,7 @@
 		/* Increase OPP mode */
 		switch (opp) {
 		case 25:
-			omap_device_set_rate(&pdev->dev, &pdev->dev, 49000000);
+			omap_device_set_rate(&pdev->dev, &pdev->dev, 98000000);
 			abe_set_opp_processing(ABE_OPP25);
 			break;
 		case 50:
